File: java-start/app/src/main/java/com/vonage/tutorial/messaging/BaseApplication.java
package com.vonage.tutorial.messaging;

import android.app.Application;
import com.vonage.tutorial.messaging.util.Tutorial;

public final class BaseApplication extends Application {

    public void onCreate() {
        super.onCreate();

        this.initializeNexmoClient();
    }

    private final void initializeNexmoClient() {
        // Init the NexmoClient. You can retrieve NexmoClient instance latter by using NexmoClient.get()
        Tutorial.todo("Init NexmoClient here");
    }
}

File: kotlin-start/app/src/main/java/com/vonage/tutorial/messaging/BaseApplication.kt
package com.vonage.tutorial.messaging

import android.app.Application

class BaseApplication : Application() {

    override fun onCreate() {
        super.onCreate()

        initializeNexmoClient()
    }

    private fun initializeNexmoClient() {
        // Init the NexmoClient. You can retrieve NexmoClient instance latter by using NexmoClient.get()
        TODO("Init NexmoClient here")
    }
}